packages/smooth_app/lib/pages/product_list_user_dialog_helper.dart

```diff
@@ -106,6 +106,59 @@ class ProductListUserDialogHelper {
     return res ?? true;
   }
 
+  /// Shows all user lists with checkboxes, adds all [barcodes] to the selected list
+  /// Filters for duplicates
+  Future<void> showBulkInsertUserListsDialog(
+    final BuildContext context,
+    final List<String> barcodes,
+  ) async {
```

Comment:
I understand the purpose, but I think we would be better off with a separate stupid method like this one:
```dart
Future<Iterable<String>?> displayUserListsDialog(
  final BuildContext context,
  final Iterable<String> initiallyCheckedUserLists,
)
```
We could reuse that dialog method every time we want to display all user lists and let the user create another one.
It will return either `null` (dialog dismissed) or a list of user lists.
Then, whatever you want can be done on those lists (e.g. add one barcode, a list of barcodes, clear those lists, whatever).
I mean, a real agnostic and reusable dialog, and then what you want to do on user lists on a different level.

Reply:
I thought about it as well but I didn't want to do any major refactorings here in this PR. Overall the whole product_list_user_dialog_helper could use some refactoring

Reply:
I wouldn't call that a refactoring as it would be a new method. I haven't even checked if it could be reused now - but probably in the product page ("add to lists").
You could code the new method now and postpone a more general refactoring by adding `TODO`s.